pwnlib/tubes/process.py

```diff
@@ -327,16 +338,20 @@ def __init__(self, argv = None,
                     args = self.argv
                     if prefix:
                         args = prefix + args
+
                     self.proc = subprocess.Popen(args = args,
-                                                 shell = shell,
-                                                 executable = executable,
-                                                 cwd = cwd,
-                                                 env = self.env,
-                                                 stdin = stdin,
-                                                 stdout = stdout,
-                                                 stderr = stderr,
-                                                 close_fds = close_fds,
-                                                 preexec_fn = self.__preexec_fn)
+                                             shell = shell,
+                                             executable = executable,
+                                             cwd = cwd,
+                                             env = self.env,
+                                             stdin = stdin if sys.platform.startswith("linux") else subprocess.PIPE,
```

Comment:
This is incorrect.

If a user manually passes in something for `stdin`, we use that `stdin`.

If you want a default fall-back, that's fine, but you have to do it elsewhere.